### PR TITLE
Pthread recursive

### PR DESCRIPTION
**Pull request by boeschf**

### CODE REVIEW

File: include/ghex/threads/omp/primitives.hpp [hunk 1]
```diff
@@ -73,6 +74,8 @@ namespace gridtools {
 
                     using mutex_type = ::gridtools::ghex::threads::mutex::pthread_spin::mutex;
                     using lock_type  = ::gridtools::ghex::threads::mutex::pthread_spin::lock_guard;
+                    //using mutex_type = ::gridtools::ghex::threads::mutex::pthread::mutex;
```
Comment (mbianco) [inline include/ghex/threads/omp/primitives.hpp]: Are these options to try? Should we guard them with macros?

Comment (boeschf) [reply include/ghex/threads/omp/primitives.hpp]: these are options which we may use for different network architectures. it turned out that for infiniband the spinlock was the most efficient. Would probably be a good idea to add macros, but we need a place where we configure the library.

Comment (mbianco) [reply include/ghex/threads/omp/primitives.hpp]: like the `config.h` mentioned by Marcin some time ago?

Comment (boeschf) [reply include/ghex/threads/omp/primitives.hpp]: yes, that would be the place I guess

**Comment by mbianco** [comment]
Can this be merged?

**Pull request closed without merging**